### ISSUE TITLE
It seems like encoding problems

### ISSUE THREAD

**Issue by ushenkodmitry** [AngleSharp/AngleSharp#324]
Hello,
I created small snippet to illustrate what do I have.
`

```
        var bc1 = BrowsingContext.New(Configuration.Default.WithDefaultLoader());
        var t1 = bc1.OpenAsync("http://www.s-cool.ru/postt292.html");
        t1.Wait();

        var bc2 = BrowsingContext.New(Configuration.Default);
        var t2 = bc2.OpenAsync(__resp => __resp.Address("http://www.s-cool.ru/postt292.html").Content(t1.Result.ToHtml()));
        t2.Wait();

        Console.WriteLine(t1.Result.Title);
        Console.WriteLine(t2.Result.Title);`
```

And the result in Console
![scool-encoding](https://cloud.githubusercontent.com/assets/16700959/14848262/8f5ba52e-0c75-11e6-9669-cd286f33e937.png)

Can this be classified as bug, or I miss something very important?

P.S. 
1. I tried every combination of configurations and cultures - the result stays the same.
2. in both cases CharacterSet detected correctly -- windows-1251

**Comment by FlorianRappl** [comment]
What you are missing is that by specifying `ToHtml` you are passing in a string, which is already converted to UTF-16. Now interpreting UTF-16 as windows-1251 may work for ASCII characters, but not for the ones you'd like to see.

If you pass in the original stream you should see that both match. Otherwise, I can suggest you to provide the encoding as UTF-16 in the second round.

The problem here is that the fake response offers both as a stream, but for a string source the encoding should be fixed to not confuse it later. Potentially this could be filed as a bug / improved, but for me the behavior is resulted from a misusage (you are entering something that was never supposed to be entered like that; either you download the page and let AngleSharp decide or you specify everything correctly).

I would now suggest the following: Try Again with the following modification.

``` cs
var bc2 = BrowsingContext.New(Configuration.Default);
var t2 = bc2.OpenAsync(__resp => __resp.Header("Content-Type", "text/html; charset=utf-8").Address("http://www.s-cool.ru/postt292.html").Content(t1.Result.ToHtml()));
t2.Wait();
```

Yes I picked UTF-8 here, but I think that should work as well. If not, please change to utf-16.

**Comment by ushenkodmitry** [comment]
Thanks for quick answer, but...
1. using utf-8 -- nothing has changed, second output is the same
2. using utf-16 -- empty string outputed

**Comment by ushenkodmitry** [comment]
There is no problem for UTF-8 charset. I meen when page defines charset via meta element, or when charset defined via Content-Type header -- everithing works correct.

**Comment by FlorianRappl** [comment]
Are you talking about _the example_ given above or _any_ other page using UTF-8?

**Comment by ushenkodmitry** [comment]
I checked on several others sources. For example http://www.ucheba.ru/. It is RU and declares UTF-8 charset.

**Comment by FlorianRappl** [comment]
Yes, but as commented if the original source is in UTF-8 (or anything other close to UTF-16) there shouldn't be any problems. What is the version of AngleSharp you are using?

I guess you are using v0.9.5 where the HTTP header has no precedence, otherwise it should work see #295.

**Comment by ushenkodmitry** [comment]
Here
package id="AngleSharp" version="0.9.5" targetFramework="net45"

**Comment by ushenkodmitry** [comment]
Assembly version 0.9.5.41771

**Comment by ushenkodmitry** [comment]
Just looked the mentioned issue. One difference -- in my case both header and meta are the same  -- windows-1251.

**Comment by FlorianRappl** [comment]
Yes, but again: The second input is a string (which is given as UTF-16). So I suppose that the issue may be already solved, but I will try tonight if I find the time. You can also build from source and check yourself it is already solved in the `devel` (pre-v0.9.6).

**Comment by ushenkodmitry** [comment]
Excellent! Solved in devel.
You thought about releasing pre or rc versions of AngleSharp packages?

**Comment by FlorianRappl** [comment]
Yes, and I think I will integrate it in the CI pipeline soon to release pre-releases after each successful build. Just didn't find the time previously to make the required changes to AppVeyor previously.

**Comment by FlorianRappl** [comment]
So I marked the issue as solved! Glad that v0.9.6 already solves your problem and I'll try to publish it soon (should be out within the next couple of days).

**Comment by sgf** [comment]
This is a bad experience. Why can't I get the corresponding encoding automatically according to the charset of the page? I still encountered this problem.
I hope to read the Html text directly from the text file, so this may not be solved by modifying the EncodingMetaHandler of the BroswingContext config.
